src/Patterns/PatternsHelper.php

```diff
@@ -122,41 +123,63 @@ public static function upsert_patterns_ai_data_post( $patterns_dictionary ) {
 	 *
 	 * @param string|null $pattern_slug The pattern slug.
 	 *
-	 * @return mixed|WP_Error|null
+	 * @return array|WP_Error Returns pattern dictionary or WP_Error on failure.
 	 */
 	public static function get_patterns_dictionary( $pattern_slug = null ) {
+		$patterns_dictionary_file = plugin_dir_path( __FILE__ ) . 'dictionary.json';
 
-		$patterns_ai_data_post = self::get_patterns_ai_data_post();
+		if ( ! file_exists( $patterns_dictionary_file ) ) {
+			return new WP_Error( 'missing_patterns_dictionary', __( 'The patterns dictionary is missing.', 'woo-gutenberg-products-block' ) );
+		}
 
-		if ( isset( $patterns_ai_data_post ) ) {
+		$default_patterns_dictionary = wp_json_file_decode( $patterns_dictionary_file, array( 'associative' => true ) );
+
+		if ( json_last_error() !== JSON_ERROR_NONE ) {
+			return new WP_Error( 'json_decode_error', __( 'Error decoding JSON.', 'woo-gutenberg-products-block' ) );
+		}
+
+		$patterns_ai_data_post = self::get_patterns_ai_data_post();
+		$patterns_dictionary   = '';
+		if ( ! empty( $patterns_ai_data_post->post_content ) ) {
 			$patterns_dictionary = json_decode( $patterns_ai_data_post->post_content, true );
-			if ( empty( $pattern_slug ) ) {
-				return $patterns_dictionary;
-			}
 
-			foreach ( $patterns_dictionary as $pattern_dictionary ) {
-				if ( $pattern_dictionary['slug'] === $pattern_slug ) {
-					return $pattern_dictionary;
-				}
+			if ( json_last_error() !== JSON_ERROR_NONE ) {
+				return new WP_Error( 'json_decode_error', __( 'Error decoding JSON.', 'woo-gutenberg-products-block' ) );
 			}
 		}
 
-		$patterns_dictionary_file = plugin_dir_path( __FILE__ ) . 'dictionary.json';
-
-		if ( ! file_exists( $patterns_dictionary_file ) ) {
-			return new WP_Error( 'missing_patterns_dictionary', __( 'The patterns dictionary is missing.', 'woo-gutenberg-products-block' ) );
+		if ( $patterns_dictionary === $default_patterns_dictionary || empty( $patterns_dictionary ) ) {
+			if ( $pattern_slug ) {
+				return self::find_pattern_by_slug( $default_patterns_dictionary, $pattern_slug );
+			}
```

Comment:
Could we remove the nested if in this way? 
```suggestion
		if ( ($patterns_dictionary === $default_patterns_dictionary || empty( $patterns_dictionary )) && $pattern_slug ) {
				return self::find_pattern_by_slug( $default_patterns_dictionary, $pattern_slug );
			}
```

Reply:
Good call! Updated on https://github.com/woocommerce/woocommerce-blocks/pull/11759/commits/0ba01c8e7850d49364c3e84850319fbbd2305b84